### PR TITLE
mon/MDSMonitor: Add confirmation to "ceph mds rmfailed"

### PR DESCRIPTION
**Pull request by ukernel**
Fixes: #14379
Signed-off-by: Yan, Zheng <zyan@redhat.com>

### CODE REVIEW

**Comment by jecluis** [comment]
lgtm, but would be nice to have this being tested by the cephtool workunit.

File: src/mon/MDSMonitor.cc [hunk 1]
```diff
@@ -1647,6 +1647,13 @@ int MDSMonitor::filesystem_command(
     }
   } else if (prefix == "mds rmfailed") {
     mds_rank_t who;
+    string confirm;
+    if (!cmd_getval(g_ceph_context, cmdmap, "confirm", confirm) ||
+	confirm != "--yes-i-really-mean-it") {
+	  ss << "WARNING: this will cause failover mechanism to stop working. "
```
Comment (jcsp) [inline src/mon/MDSMonitor.cc]: We should make this message more severe.

Maybe "WARNING: this can make your filesystem inaccessible, use extreme caution!"

**Comment by gregsfortytwo** [comment]
While we're making changes we should add the checks and warnings to any other commands which deserve them — like the recently demonstrated setmap command. ;)

**Comment by gregsfortytwo** [comment]
My testing branch is failing make check with "FAIL: test/pybind/test_ceph_argparse.py", and I presume it's because this PR or https://github.com/ceph/ceph/pull/7262 got broken somehow.

**Comment by ukernel** [comment]
<pre>
bluestore_wal_transaction_t
/tmp/typ-ybpvQW9mL /tmp/typ-eV5I52Jt8 differ: byte 60, line 5
**** bluestore_wal_transaction_t test 2 dump_json check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 dump_json > /tmp/typ-ybpvQW9mL
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 encode decode dump_json > /tmp/typ-eV5I52Jt8
   diff /tmp/typ-ybpvQW9mL /tmp/typ-eV5I52Jt8
/tmp/typ-ybpvQW9mL /tmp/typ-3q8FRszW0 differ: byte 61, line 5
**** bluestore_wal_transaction_t test 2 copy dump_json check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 dump_json > /tmp/typ-ybpvQW9mL
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 copy dump_json > /tmp/typ-eV5I52Jt8
   diff /tmp/typ-ybpvQW9mL /tmp/typ-eV5I52Jt8
/tmp/typ-ybpvQW9mL /tmp/typ-ptIzfCk06 differ: byte 60, line 5
**** bluestore_wal_transaction_t test 2 copy_ctor dump_json check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 dump_json > /tmp/typ-ybpvQW9mL
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 copy_ctor dump_json > /tmp/typ-eV5I52Jt8
   diff /tmp/typ-ybpvQW9mL /tmp/typ-eV5I52Jt8
/tmp/typ-ybpvQW9mL /tmp/typ-eV5I52Jt8 differ: byte 25, line 1
**** bluestore_wal_transaction_t test 2 binary reencode check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 encode export /tmp/typ-ybpvQW9mL
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 encode decode encode export /tmp/typ-eV5I52Jt8
   cmp /tmp/typ-ybpvQW9mL /tmp/typ-eV5I52Jt8
</pre>

the failure of 'make check' should not be related to this change

**Comment by gregsfortytwo** [comment]
I know the bluestore encode failures are unrelated, but there are also failures in ceph_test_argparse around the commands being changed in this and the companion PR.

**Comment by ukernel** [comment]
ceph_test_argparse  failure has already been fixed

**Pull request merged**